kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveGlobalStore.java

```diff
@@ -93,8 +89,12 @@ public void init(final StateStoreContext context, final StateStore root) {
       // to be more evently distributed and take the hit only on range
       // queries
       partition = 0;
+      final SharedClients sharedClients = new SharedClients(context.appConfigs());
+      client = sharedClients.cassandraClient;
+
+      final RemoteMonitor monitor = client.awaitTable(name.cassandraName(), sharedClients.executor);
```

Comment:
the nice thing about sharing this across threads is that if there are many partitions only one of them needs to satisfy the condition and then we'll stop querying scylla and we're not hammering the server with metadata requests

thoughts on adding this to the shared clients?

Reply:
I actually did think about it! Glad you called this out, I meant to leave a comment asking you but forgot...I was wondering if this would have any nontrivial side effect or is more of a nice-to-have optimization?

The trouble is that with the new pattern, the first time we get access to both the `cassandraName` (ie store name) and the configs (with shared clients) is here in `#init`. So we can't do this any earlier, right?

Beyond the scope of this particular PR, but I was wondering if we could instead move the table creation outside of the state stores entirely -- for example, after creating the other shared clients in  `ResponsiveKafkaStreams` we would also parse the `Topology` for all the store names and create a RemoteMonitor for each one to stash alongside the other shared clients. Does that make sense? Maybe a followup PR?

Reply:
I see that makes sense, let's punt the optimization to a future PR. Not urgent for now.

Reply:
I filed a ticket to revisit this https://www.notion.so/responsivedev/Create-C-tables-during-shared-client-creation-b28c4c54f3e745e4a7b4a1a4ef2d3294?pvs=4